proto/redpanda/core/admin/v2/shadow_link.proto

```diff
@@ -260,6 +260,17 @@ message TopicMetadataSyncOptions {
     // If this is true, then only the properties listed in
     // `synced_shadow_topic_properties` will be synced.
     bool exclude_default = 4;
+    // Start at the earliest offset in the partition
+    message EarliestOffset {}
+    // Start at the latest offset in the partition
+    message LatestOffset {}
+    // The starting offset for new shadow topics.  If none are set, defaults to
+    // earliest
+    oneof start_offset {
+        EarliestOffset earliest = 5;
+        LatestOffset latest = 6;
+        google.protobuf.Timestamp timestamp = 7;
```

Comment:
nit: add some docs for this doing a timequery and start from the earliest record that starts at or after this timestamp

Reply:
done.